mobile/src/main/java/org/openhab/habdroid/ui/preference/CustomInputTypePreference.kt

```diff
@@ -67,26 +69,52 @@ open class CustomInputTypePreference constructor(context: Context, attrs: Attrib
         return PrefFragment.newInstance(key, title, inputType, autofillHints)
     }
 
-    class PrefFragment : EditTextPreferenceDialogFragmentCompat() {
+    class PrefFragment : EditTextPreferenceDialogFragmentCompat(), TextWatcher {
+        private lateinit var wrapper: TextInputLayout
+        private lateinit var editor: EditText
+
         override fun onBindDialogView(view: View?) {
-            val wrapper = view?.findViewById<TextInputLayout>(R.id.input_wrapper)
-            val editor = view?.findViewById<EditText>(android.R.id.edit)
-            arguments?.getInt(KEY_INPUT_TYPE)?.let { type ->
-                editor?.inputType = type
-            }
-            arguments?.getCharSequence(KEY_TITLE)?.let { title ->
-                wrapper?.hint = title
-            }
-            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
-                val hints = arguments?.getStringArray(KEY_AUTOFILL_HINTS)
-                if (hints == null) {
-                    editor?.importantForAutofill = View.IMPORTANT_FOR_AUTOFILL_NO
-                } else {
-                    editor?.importantForAutofill = View.IMPORTANT_FOR_AUTOFILL_YES
-                    editor?.setAutofillHints(*hints)
+            super.onBindDialogView(view)
+            if (view != null) {
```

Comment:
Better check for `view == null` and return if so.

Reply:
Or keep initialization of `editor` and `wrapper` as they are (can keep them lateinit then) and just return if they're null.